src/api/unified/signal.cpp

```diff
@@ -13,22 +13,36 @@
 
 af_err af_approx1(af_array *yo, const af_array yi, const af_array xo,
                   const af_interp_type method, const float offGrid) {
-    CHECK_ARRAYS(yi, xo);
+    CHECK_ARRAYS(yo, yi, xo);
+    return CALL(yo, yi, xo, method, offGrid);
+}
+
+af_err af_approx1_v2(af_array *yo, const af_array yi, const af_array xo,
+                     const af_interp_type method, const float offGrid) {
+    CHECK_ARRAYS(yo, yi, xo);
     return CALL(yo, yi, xo, method, offGrid);
 }
 
 af_err af_approx2(af_array *zo, const af_array zi, const af_array xo,
                   const af_array yo, const af_interp_type method,
                   const float offGrid) {
-    CHECK_ARRAYS(zi, xo, yo);
+    CHECK_ARRAYS(zo, zi, xo, yo);
```

Comment:
My changes for adding the _v2 versions of `af_approx2`/`af_approx2_uniform` are still underway (no PR yet). Should we include this change in this PR now? Alternately, I can add them in the changes that I'm working on right now.

Reply:
This is working on the original function. you don't have to update that in your PR. You are going to be adding the v2 functions.